batavia/types/Str.js

```diff
@@ -233,7 +233,7 @@ String.prototype.__iadd__ = function(other) {
 };
 
 String.prototype.__isub__ = function(other) {
-    throw new batavia.builtins.NotImplementedError("String.__isub__ has not been implemented");
+    throw new batavia.builtins.TypeError("unsupported operand type(s) for -=: 'str' and '" + batavia.type_name(other) +"'")
```

Comment:
Looking at e916704fdb4a9c629b86bdb06324509ec3c099d2, they've implemented a series of if statements to catch different types of "bad" input. This is great, because it means we can append to the work they've done or replace their `if` statement when we're ready to finish that function.

Since your work is doing similar things, it would be great if it followed a similar pattern. Following that pattern will make it easier to edit later, and be closer to the complete solution.